src/components/user-profile.html

```diff
@@ -1,111 +1,256 @@
+        text-decoration: none;
+        display: block;
+        margin: auto;
+        font-size: 16px;
+        margin-top: 20px;
+        margin-bottom: 20px;
+        border-radius: 8px;
+      }
+
+      button:hover {
+        background-color: rgb(196, 226, 245);
+      }
+
+      table.center {
+        margin-top: 20px;
+        margin-right: auto;
+        margin-left: auto;
+      }
+
+      th,
+      td {
+        text-align: left;
+        padding: 16px;
+      }
+
+      tr:nth-child(odd) {
+        background-color: #f2f2f2;
+      }
+
+      .container {
+        padding: 0 16px;
+        display: flex;
+      }
+
+      .container::after,
+      .row::after {
+        content: "";
+        clear: both;
+        display: table;
+      }
+
+      .card {
+        padding: 5px;
+        box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
+        border-radius: 20px;
+      }
+
+      /* Edit Form */
+      .edit-div {
+        border-radius: 5px;
+        background-color: #f2f2f2;
+        margin: 25px auto 25px auto;
+        width: 60%;
+      }
+
+      .edit-container {
+        padding: 10px 16px;
+        width: 60%;
+      }
+
+      .edit-Card {
+        box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
+        padding-top: initial;
+      }
+
+      .edit-container::after,
+      .row::after {
+        content: "";
+        clear: both;
+        display: table;
+      }
+
+      input[type=text],
+      select {
+        width: 100%;
+        padding: 12px 20px;
+        margin: 8px 0;
+        display: inline-block;
+        border: 1px solid #87CEFA;
+        border-radius: 4px;
+        box-sizing: border-box;
+      }
+
+    </style>
+    <button class="button" on-click="expandCollapseCard">[[buttonText]]</button>
+
+    <template is="dom-if" if="[[!editButtonClicked]]">
+      <div class="div-center" id="userProfileCard">
+        <div class="card">
+          <h3 class="div-center">User Profile</h3>
+
+          <table class="container center" id="test">
+            <tr>
+              <td>User Id:</td>
+              <td>[[user.id]]</td>
+            </tr>
+            <tr>
+              <td>Name:</td>
+              <td>[[user.firstName]] [[user.lastName]]</td>
+            </tr>
+
+            <template is="dom-if" if="[[profileExpanded]]">
+              <tr>
+                <td>Phone:</td>
+                <td>[[user.phone]]</td>
+              </tr>
+              <tr>
+                <td>Email:</td>
+                <td>[[user.email]]</td>
+              </tr>
+              <button class="button" on-click="editButtonClick">Edit</button>
+              <button class="button" on-click="deleteUser">Delete</button>
+              <button class="button" on-click="expandCollapseCard">Collapse</button>
+            </template>
+          </table>
+        </div>
+      </div>
+    </template>
+
+    <template is="dom-if" if="[[editButtonClicked]]">
+        <div class="edit-div edit-container edit-card">
```

Comment:
nit: this section between the `<template is="dom-if" if="[[editButtonClicked]]">` could be moved into your edit user component, the component imported into `user-profile`, and used here rather hardcoding it within the user profile component itself.